demo/templates/demo/analytics.html

```diff
@@ -1,4 +1,7 @@
 <script async defer src="https://sa.mathesar.org/latest.js"></script>
+<script>
+    window.addEventListener("event", (e) => sa_event(e.detail.eventName, e.detail.metadata));
+</script>
```

Comment:
`sa_event` would be undefined until the async analytics script loads, so the function will throw.

Event listeners ignore any errors within them, so we don't have to worry about a DOM wide error. Also, we don't dispatch any event during that time so we probably wouldn't have any lost events.

I would still like some run time checks to ensure `sa_event` is defined, and `e.detail` is defined, and print a console error if not, to ensure we're aware of any future issues during development. Since we can't use typescript here, I feel that a runtime check would be good. This is a pattern I'd like to discuss @seancolsen @rajatvijay 

@silentninja No changes needed in this PR. If we do introduce some pattern around this, we'll take that up later.